plugins/org.komodo.relational/src/org/komodo/relational/internal/RelationalObjectImpl.java

```diff
@@ -34,6 +35,19 @@ protected RelationalObjectImpl( final Repository repository,
     /**
      * {@inheritDoc}
      *
+     * @see org.komodo.repository.ObjectImpl#setPrimaryType(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     * @throws UnsupportedOperationException
+     *         if called
+     */
+    @Override
+    public final void setPrimaryType( final UnitOfWork uow,
+                                      final String typeName ) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
```

Comment:
Why is this throwing an UOE?
UOE is a runtime exception so does not need a throws clause. Also, the fact it does not need a throws clauses means it catching is undetermined. Can we replace it with a normal exception if it should be throwing an exception.